src/component/mxgraph/StyleConfigurator.ts

```diff
@@ -144,9 +146,14 @@ export default class StyleConfigurator {
     this.putCellStyle(ShapeBpmnElementKind.GATEWAY_PARALLEL, style);
   }
 
-  private configureExclusiveGatewayStyle(): void {
-    const style = this.mxUtils.clone(this.getStylesheet().getCellStyle(ShapeBpmnElementKind.GATEWAY_PARALLEL), this.getDefaultVertexStyle());
-    style[this.mxConstants.STYLE_GRADIENTCOLOR] = '#DDA0DD';
-    this.putCellStyle(ShapeBpmnElementKind.GATEWAY_EXCLUSIVE, style);
+  private configureGatewaysStyle(): void {
+    ShapeUtil.topLevelBpmnGatewayKinds().forEach(kind => {
+      const style = this.cloneDefaultVertexStyle();
+      style[this.mxConstants.STYLE_SHAPE] = kind;
+      style[this.mxConstants.STYLE_PERIMETER] = this.mxPerimeter.RhombusPerimeter;
+      style[this.mxConstants.STYLE_VERTICAL_ALIGN] = 'top';
+
+      this.putCellStyle(kind, style);
```

Comment:
Restore the hard coded label positioning as it was before
- no change in the label rendering compared to the previous implementation
- move the label out of the shape --> we can identify the gateway type, otherwise the label hide the gateway icon

```suggestion
      // TODO to be removed when supporting label position
      // left just to not break current rendering
      style[this.mxConstants.STYLE_SPACING_TOP] = 55;
      style[this.mxConstants.STYLE_SPACING_RIGHT] = 110;

      this.putCellStyle(kind, style);
```